### PR TITLE
chore: cypress and travis ci

### PR DESCRIPTION
**Pull request by mereck**
# Changes

- Adds cypress
- Adds start-server-and-test
- Adds travis config
- Adds ci-related test scripts to package.json

#References
#76

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,6 +36,7 @@
     "sites:launch-test": "npm run sites:clone-local test-site && npm run sites:update test-site && cd sites/test-site && npm install && npm run start",
     "start": "npm run build:watch",
     "test": "jest",
+    "test:ci": "start-server-and-test start http://localhost:8005 cy:run-parallel",
```
Comment (wodenx) [inline package.json]: For greater fidelity, we should run these tests using local packages - ie, do 'sites:clone-local && sites:update test-site && cd sites/test-site && npm i && npm run start as the start command.

Comment (mereck) [reply package.json]: Thanks - I have updated the script to use `sites:launch-test` instead of `start`

File: package.json [hunk 2]
```diff
@@ -13,6 +13,9 @@
     "build:watch": "cross-env NODE_ENV=development lerna run build:watch --parallel",
     "clean": "lerna run clean && rimraf .eslintcache examples/*/.env.development examples/*/.env.production",
     "copyright": "node ./scripts/add-copyright",
+    "cy:run": "cypress run",
```
Comment (TatsianaMiraniuk) [inline package.json]: need to specify '--browser chrome'
by default it's running in Electron.

Comment (TatsianaMiraniuk) [inline package.json]: in 'run' command we need to specify '--browser chrome'
by default it's running in Electron.

**Comment by TatsianaMiraniuk** [comment]
> @mereck and @TatsianaMiraniuk I am approving bc I want the CI piece merged, but I think there are som improvements to the spec itself which should be pulled into another ticket. @TatsianaMiraniuk please review and create the issue (in github).

@wodenx Created https://github.com/johnsonandjohnson/Bodiless-JS/issues/92. Agree that this test should not block us from merging. 
@mereck please see my comment regarding specifying a browser.

File: cypress/integration/link_menu.spec.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
+/* eslint-disable jest/valid-expect */
+/* eslint-disable no-undef */
+// <reference types="Cypress" />
+
+context('Link Context Menu', () => {
+  let nodeKey = '';
+  const backendPath = '/___backend/content/pages/context';
+
+  beforeEach(() => {
+    cy.visit('/context').wait(1000);
```
[the remark this answers is not in the document]
Comment (mereck) [reply cypress/integration/link_menu.spec.js]: not necessary anymore, will remove

File: cypress/integration/link_menu.spec.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    const someText = Math.random().toString();
+
+    cy.get('@form-popup')
+      .find('input#link-href')
+      .as('link-href')
+      .clear()
+      .type(someText);
+
+    cy.server();
+    cy.route('POST', `${backendPath}/*`).as('backend');
```
[the remark this answers is not in the document]
Comment (mereck) [reply cypress/integration/link_menu.spec.js]: nope, this is an alias to an actual backend endpoint, which allows the `cy.wait('@backend')` below to inspect the actual request

File: package.json (hunk 2 above)
Comment (mereck) [reply package.json]: good point, thanks

File: cypress/integration/link_menu.spec.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+    cy.get('@form-popup')
+      .find('form')
+      .submit();
+
+    cy.wait('@backend').should(xhr => {
+      expect(xhr.request.body).to.haveOwnProperty('href', someText);
+      // @TODO: need a way to tie a component node-key to markup
+      // e.g. <a href='...' data-node-key="linkit" >some link</a>
+      // otherwise there is currently no way to validate if correct backend endpoint was hit
```
[the remark this answers is not in the document]
Comment (mereck) [reply cypress/integration/link_menu.spec.js]: #94

[the remark this answers is not in the document]
Comment (mereck) [reply cypress/integration/link_menu.spec.js]: Agreed on full e2e interaction including file system.

File: cypress/integration/link_menu.spec.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+context('Link Context Menu', () => {
+  let nodeKey = '';
+  const backendPath = '/___backend/content/pages/context';
+
+  beforeEach(() => {
+    cy.visit('/context').wait(1000);
+  });
+
+  it('should retain submitted value when reopened', () => {
+    cy.get('div.flex-1')
```
[the remark this answers is not in the document]
Comment (mereck) [reply cypress/integration/link_menu.spec.js]: Agreed, although I am not sure how to improve them without changing the markup - we don't have any unique ids there.

File: package.json [hunk 7]
```diff
@@ -14,6 +14,9 @@
     "build:env-vars": "cross-env NODE_ENV=production lerna run build:env-vars --stream",
     "clean": "lerna run clean && rimraf .eslintcache examples/*/.env.development examples/*/.env.production",
     "copyright": "node ./scripts/add-copyright",
+    "cy:run": "cypress run --browser chrome",
+    "cy:run-parallel": "cypress run --record --key cf8146aa-c72d-4767-83e1-29d354f0f453 --parallel --group $STAGE_NAME",
```
Comment (TatsianaMiraniuk) [inline package.json]: @mereck can we also add '--browser chrome' in cy:run-parallel as that command triggers the tests?

Comment (mereck) [reply package.json]: done

File: package.json [hunk 8]
```diff
@@ -13,6 +13,7 @@
     "build:watch": "cross-env NODE_ENV=development lerna run build:watch --parallel",
     "build:env-vars": "cross-env NODE_ENV=production lerna run build:env-vars --stream",
     "clean": "lerna run clean && rimraf .eslintcache examples/*/.env.development examples/*/.env.production",
+    "cy:run": "cypress run --browser chrome --record --key 0dcfb624-e152-453a-9375-aa710adb57ab",
```
Comment (TatsianaMiraniuk) [inline package.json]: @mereck I believe we should remove '--record --key 0dcfb624-e152-453a-9375-aa710adb57ab' as that is used for recording results in Dashboard in your public project.

File: cypress/integration/link_menu.spec.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (wodenx) [reply cypress/integration/link_menu.spec.js]: I think we need a ticket to improve this test case, including changing the markup to support better selectors.  @TatsianaMiraniuk  Can you create one?
[ ] Create ticket.

[the remark this answers is not in the document]
Comment (wodenx) [reply cypress/integration/link_menu.spec.js]: #92

File: cypress.json [hunk 9]
```diff
@@ -0,0 +1,4 @@
+{
+  "projectId": "totsvs",
```
[the remark this answers is not in the document]
Comment (TatsianaMiraniuk) [reply cypress.json]: @mereck please replace to this value: d24g6k

File: package.json (hunk 8 above)
Comment (TatsianaMiraniuk) [reply package.json]: @mereck please update the key to this value: a5bd2109-f9d9-4533-9fc7-b54c05c691b8

File: cypress.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (TatsianaMiraniuk) [reply cypress.json]: keep the old id as the project was transferred

File: package.json (hunk 8 above)
Comment (TatsianaMiraniuk) [reply package.json]: decided to transfer the initial project so we keep the key

**Pull request merged**